lib/datadog/profiling/load_native_extension.rb

```diff
@@ -18,7 +18,20 @@
 end
 
 extension_name = "datadog_profiling_native_extension.#{RUBY_VERSION}_#{RUBY_PLATFORM}"
-full_file_path = "#{__dir__}/../../#{extension_name}.#{RbConfig::CONFIG['DLEXT']}"
+file_name = "#{extension_name}.#{RbConfig::CONFIG['DLEXT']}"
+full_file_path = "#{__dir__}/../../#{file_name}"
```

Comment:
Indeed... The profiler really only targets Linux/Unix right now, so a number of linux/unixisms have crept in.

The native extension loader that's getting called using this path, for instance, uses `dlopen`, so that's another thing that wouldn't work.

So yeah, I'm not sure it's worth trying to piecemeal support all OSs for the profiler in preparation for maybe supporting them in the future, I think it's easier to evaluate the whole thing once we want that (e.g. for instance not using the `load_native_extension.rb` on Windows at all).